OneSignalSDK/onesignal/src/main/java/com/onesignal/OSInAppMessageController.java

```diff
@@ -189,19 +190,36 @@ private void processInAppMessageJson(@NonNull JSONArray json) throws JSONExcepti
         }
         messages = newMessages;
 
+        populateRedisplayMessagesTriggers();
         evaluateInAppMessages();
     }
 
+    private void populateRedisplayMessagesTriggers() {
+        for (OSInAppMessage redisplayMessage : redisplayedInAppMessages) {
```

Comment:
This logic seems odd to me by using a foreach style logic but then looking up an idex. Couldn't this work with a `contains` check instead?

Reply:
we need the index in order to update, nevertheless, this logic can be improved by using the processInAppMessageJson loop

Reply:
👍 This make it easier to read